src/Controller.ino

```diff
@@ -246,17 +246,17 @@ bool MQTTConnect(controllerIndex_t controller_idx)
   log += subscribeTo;
   addLog(LOG_LEVEL_INFO, log);
 
-  if (MQTTclient.publish(LWTTopic.c_str(), LWTMessageConnect.c_str(), 1)) {
-    updateMQTTclient_connected();
-    statusLED(true);
-    mqtt_reconnect_count = 0;
+  updateMQTTclient_connected();
+  statusLED(true);
+  mqtt_reconnect_count = 0;
 
-    // call all installed controller to publish autodiscover data
-    if (MQTTclient_should_reconnect) { CPluginCall(CPlugin::Function::CPLUGIN_GOT_CONNECTED, 0); }
-    MQTTclient_should_reconnect = false;
-    return true; // end loop if succesfull
-  }
-  return false;
+  // call all installed controller to publish autodiscover data
+  if (MQTTclient_should_reconnect) { CPluginCall(CPlugin::Function::CPLUGIN_GOT_CONNECTED, 0); }
+  MQTTclient_should_reconnect = false;
+
+  if (ControllerSettings.mqtt_sendLWT()) { MQTTclient.publish(LWTTopic.c_str(), LWTMessageConnect.c_str(), willRetain); }
```

Comment:
I think the publish of the LWT should be before the call to CPLUGIN_GOT_CONNECTED

Reply:
Hmm on second thought, it may be better not to let this publish decide on whether it is connected or not.
The publish may fail for various reasons and there have been reports of lots of MQTT reconnect attempts on some systems.

Maybe this LWT publish state should be dealt with asynchronous, and not let the connected status depend on it.

Reply:
Hmm I'm looking into the PubSubClient code and I wonder why we even try to publish to the LWT topic, as that's already been done by the connect itself.

Reply:
I figured successful connection cannot be determined by the connect message, given the connect message should be optional.

How critical is the "Connected" status message? It's useful for troubleshooting, but are automated processes likely to depend on it? Probably not, but that's just me.

Currently with the "Send LWT" setting disabled, a client will disconnect silently leaving the retained "Connected" LWT message, which is quite misleading.

And; GitKraken has been a revalation, thanks! Much easier to make sense of what is going on than CLI or even SourceTree :-)

Reply:
I just pushed an extra commit to this PR, but based on your comment I think I still need to make a simple addition to stop trying to send the "connected" LWT message when the client is disconnected.

It is possible to flag the LWT connected message with the retain flag, so it can be used in some systems to get a quick view of who's online.
So in some systems it may be quite critical, but I don't think the return status of the connect attempt should depend on the success of the call to publish the LWT connect message.

Reply:
Fair enough; makes sense to me.